server/src/main/java/org/apache/druid/client/cache/ForegroundCachePopulator.java

```diff
@@ -75,21 +80,19 @@ public ForegroundCachePopulator(
         Sequences.map(
             sequence,
             input -> {
-              if (!tooBig.get()) {
-                synchronized (lock) {
```

Comment:
This lock removal is the core change happening, and it looks good to me. The lock seems to have been pointless, since the only thing worth protecting is the `ByteArrayOutputStream`, and that's only being accessed by a single thread (the thread that walks the Sequence).